src/main/java/org/opensearch/flowframework/model/WorkflowState.java

```diff
@@ -196,8 +223,8 @@ public Builder userOutputs(Map<String, Object> userOutputs) {
          * @param resourcesCreated resourcesCreated
          * @return the Builder object
          */
-        public Builder resourcesCreated(Map<String, Object> resourcesCreated) {
-            this.userOutputs = resourcesCreated;
+        public Builder resourcesCreated(List<ResourcesCreated> resourcesCreated) {
+            this.resourcesCreated = resourcesCreated;
```

Comment:
what if there's no created resources? Or will the caller pass an empty list if so?

Reply:
We create the state here, user wont ever change the resources created, I initiate it with an empty list though so it takes care of that case.